packages/components/link/src/link.tsx

```diff
@@ -104,21 +104,28 @@ const getActiveColorValue = (tone: string = 'primary') => {
   return designTokens.fontColorForTextWhenInverted;
 };
 
-const getLinkStyles = (props: TLinkProps) => {
+const getLinkStyles = (props: TLinkProps & { isNewTheme: boolean }) => {
   const color = getTextColorValue(props.tone);
   const hoverColor = getActiveColorValue(props.tone);
 
-  return css`
-    font-family: inherit;
-    color: ${color};
-    font-size: ${designTokens.fontSizeForLink};
-    &:hover,
-    &:focus,
-    &:active {
-      color: ${hoverColor};
-    }
-    text-decoration: underline;
-  `;
+  return [
+    css`
+      font-family: inherit;
+      color: ${color};
+
+      &:hover,
+      &:focus,
+      &:active {
+        color: ${hoverColor};
+      }
+      text-decoration: underline;
+    `,
+    !props.isNewTheme
+      ? css`
+          font-size: ${designTokens.fontSizeForLink};
+        `
+      : null,
+  ];
```

Comment:
I tried using different values for the new theme, but the one that would work (`unset`) can't be used in our context because of the css rule that gets created:
```
var(--font-size-for-link, 1rem)
```

When the value is `unset`, the `var` call uses the default value (`1rem`)